--- OgreCodec.Unchanged.h	2012-10-06 16:15:43.280415476 -0500
+++ OgreCodec.h	2012-10-05 20:13:29.165702696 -0500
@@ -35,6 +35,12 @@
 #include "OgreStringVector.h"
 #include "OgreException.h"
 
+#if __GNUC__ >= 4
+    #define DLL_LOCAL  __attribute__ ((visibility ("hidden")))
+#else
+    #define DLL_LOCAL
+#endif
+
 namespace Ogre {
 	/** \addtogroup Core
 	*  @{
@@ -58,7 +64,7 @@
         typedef map< String, Codec* >::type CodecList; 
         /** A map that contains all the registered codecs.
         */
-        static CodecList msMapCodecs;
+        DLL_LOCAL static CodecList msMapCodecs;
 
     public:
         class _OgrePrivate CodecData : public CodecAlloc
