--- SDL_video.original.c	2013-04-01 15:19:12.061967916 -0500
+++ SDL_video.c	2013-04-01 15:19:38.117968182 -0500
@@ -1418,7 +1418,7 @@
     /* See if the named data already exists */
     prev = NULL;
     for (data = window->data; data; prev = data, data = data->next) {
-        if (SDL_strcmp(data->name, name) == 0) {
+        if (data->name && SDL_strcmp(data->name, name) == 0) {
             void *last_value = data->data;
 
             if (userdata) {
