+++ "b/D:\\MezzanineSafe\\Mezzanine\\libincludes\\common\\openalsoftsrc\\openalsoft\\CMakeLists.txt"
@@ -279,9 +279,13 @@ 
 CHECK_INCLUDE_FILE(fenv.h HAVE_FENV_H)
 CHECK_INCLUDE_FILE(float.h HAVE_FLOAT_H)
 CHECK_INCLUDE_FILE(ieeefp.h HAVE_IEEEFP_H)
 CHECK_INCLUDE_FILE(guiddef.h HAVE_GUIDDEF_H)
+CHECK_INCLUDE_FILE(pthread.h HAVE_PTHREAD_H)
 IF(NOT HAVE_GUIDDEF_H)
     CHECK_INCLUDE_FILE(initguid.h HAVE_INITGUID_H)
 ENDIF()
+IF(HAVE_PTHREAD_H)
+    ADD_DEFINITIONS(-DHAVE_PTHREAD_H)
+ENDIF()
 
 # Some systems need libm for some of the following math functions to work
 CHECK_LIBRARY_EXISTS(m pow "" HAVE_LIBM)
